### PR TITLE
Add content to README; documentation fixups

### LINKED ISSUE #886: Update README/docs to better reflect the purpose of Maggma
### Problem

In talking to colleagues unfamiliar with maggma, when they check out the repo, it seems there is some (justifiable) confusion about the main purpose of the package.

For instance, the "About" section on the GitHub sidebar says "MongoDB aggregation machine", which of course is how the name was derived, but it leads visitors to believe that it is a package only for MongoDB when in reality there are many data stores.

The README says "A files-to-API data pipeline," but in fact, many of the data stores do not rely on the filesystem at all. The same point can be made for the "What is Maggma" section of [the documentation](https://materialsproject.github.io/maggma/#what-is-maggma).

In general, it is not immediately clear the purpose of Maggma from the docs, and there is a lot of jargon once you dive into the docs.

### Proposed Solution

1. Come up with some clearer branding on the README and "What is Maggma" section.
2. Add a "quickstart" section to the Maggma docs, before the Core Concepts, that shows a common minimal example. For instance, instantiate a `MongoStore`, upload a toy result, and query it.

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by rkingsbury**
## Summary

Makes `README` more descriptive of `maggma''s purpose and function. Closes #886 

See a preview of the README on my fork here:
https://github.com/rkingsbury/maggma/tree/readme

### CODE REVIEW

**Comment by rkingsbury** [comment]
@Andrew-S-Rosen any suggested revision are welcome!

@munrojm does it make sense to add a paragraph about what the API piece of `maggma` does? If so, could you post some text here and I can integrate it?

Looking at the docs, I realize that there is currently nothing in there about the API components at all; that's probably something we should add at some point (separate issue though)

**Comment by Andrew-S-Rosen** [comment]
Looks great! Did we want to add a minimal example so people visually know what we are talking about? Something like

```python
my_data = {"a": 1, "b": 2}
store = MongoStore( database="my_db_name", collection_name="my_collection_name", username="my_username", password="my_password", host="my_hostname", port=27017, ) 
with store:
        store.update(my_data)
```

File: README.md [hunk 1]
```diff
@@ -39,6 +39,15 @@ s2 -- Builder 3-->s4(Store 4)
 
 A major challenge in building scalable data pipelines is dealing with all the different types of data sources out there. Maggma's `Store` class provides a consistent, unified interface for querying data from arbitrary data sources. It was originally built around MongoDB, so it's interface closely resembles `PyMongo` syntax. However, Maggma makes it possible to use that same syntax to query other types of databases, such as Amazon S3, GridFS, or files on disk, [and many others](https://materialsproject.github.io/maggma/getting_started/stores/#list-of-stores). Stores implement methods to `connect`, `query`, find `distinct` values, `groupby` fields, `update` documents, and `remove` documents.
 
+```python
+>>> my_data = {"a": 1, "b": 2}
+>>> store = MongoStore(database="my_db_name", collection_name="my_collection_name", username="my_username", >>> password="my_password", host="my_hostname", port=27017, )
```
Comment (Andrew-S-Rosen) [inline README.md]: Can probably remove the stray `, )` at the end of the `MongoStore` args. Sorry, was typing on my phone :)

**Comment by Andrew-S-Rosen** [comment]
Looks good to me!

**Comment by Andrew-S-Rosen** [comment]
That example is hilarious!

**Comment by rkingsbury** [comment]
Now that I realize we don't have any documentation of the API components, I'm going to remove the placeholder from the README and merge this. I've opened #972 as a placeholder to remind us that this documentation should be added at some point.

**Comment by rkingsbury** [comment]
There is an open_data test failure that snuck in on the last dependency upgrade, which I can't reproduce locally. I'm merging this b/c it's unrelated and will see if re-triggering the auto dependency upgrade can resolve.

**Pull request merged**